tools/schemacode/src/bidsschematools/types/namespace.py

```diff
@@ -272,10 +272,13 @@ def from_json(cls, jsonstr: str):
 def _read_yaml_dir(path: Path) -> dict:
     mapping = {}
     for subpath in sorted(path.iterdir()):
-        if subpath.is_dir():
-            mapping[subpath.name] = _read_yaml_dir(subpath)
-        elif subpath.name.endswith("yaml"):
-            mapping[subpath.stem] = yaml.safe_load(subpath.read_text())
+        try:
+            if subpath.is_dir():
+                mapping[subpath.name] = _read_yaml_dir(subpath)
+            elif subpath.name.endswith("yaml"):
+                mapping[subpath.stem] = yaml.safe_load(subpath.read_text())
+        except Exception as e:
+            raise ValueError(f"There was an error reading the file: {subpath}") from e
```

Comment:
Rather than creating a chain of exceptions as long as the file is dep, we can just get one:

```suggestion
        if subpath.is_dir():
            mapping[subpath.name] = _read_yaml_dir(subpath)
        elif subpath.name.endswith("yaml"):
            try:
                mapping[subpath.stem] = yaml.safe_load(subpath.read_text())
            except Exception as e:
                raise ValueError(f"There was an error reading the file: {subpath}") from e
```

Reply:
yes indeed!!